lib/magicbell-rails.rb

```diff
@@ -11,5 +11,22 @@ module Rails
     def self.bell(args)
       Notification.bell(args)
     end
+
+    def self.fetch_categories(external_id)
+      response = client.user_with_external_id(external_id).notification_preferences.retrieve.attributes
+
+      response['categories']&.map { |category| category['slug'] } || []
+    end
+
+    def self.update_notification_preferences(external_id, payload)
+      client.user_with_external_id(external_id).notification_preferences.update(payload)
+    end
```

Comment:
suggestion pushing the logic deeper

```suggestion
    def self.update_notification_preferences(external_id, payload)
      NotificationPerference.update(external_id, payload)
    end
```